numpy/core/function_base.py

```diff
@@ -279,7 +280,9 @@ def logspace(start, stop, num=50, endpoint=True, base=10.0, dtype=None,
     >>> plt.show()
 
     """
+    start, stop, base = np.broadcast_arrays(start, stop, base, subok=True)
```

Comment:
In principle, if `base` has many elements but `start` and `stop` do not, this will cause a lot of extra calculation for no benefit. Slightly better but a bit more annoying,
```
ndmax = np.broadcast(start, stop, base).ndim
start, stop, base = (np.array(a, copy=False, subok=True, ndmin=ndmax) for a in (start, stop, base))
```